trace-utils/src/trace_utils.rs

```diff
@@ -82,9 +84,15 @@ pub fn coalesce_send_data(mut data: Vec<SendData>) -> Vec<SendData> {
     data.sort_unstable_by(|a, b| a.target.url.to_string().cmp(&b.target.url.to_string()));
     data.dedup_by(|a, b| {
         if a.target.url == b.target.url {
-            a.tracer_payloads.append(&mut b.tracer_payloads);
-            a.size += b.size;
-            return true;
+            // Size is only an approximation. In practice it won't vary much, but be safe here.
+            // We also don't care about the exact maximum size, like two 25 MB or one 50 MB request
+            // has similar results. The primary goal here is avoiding many small requests.
+            // TODO: maybe make the MAX_PAYLOAD_SIZE configurable?
+            if a.size + b.size < MAX_PAYLOAD_SIZE / 2 {
+                b.tracer_payloads.append(&mut a.tracer_payloads);
+                b.size += a.size;
```

Comment:
Maybe call out the order of `dedup_by` in the comment so we don't break this again?

Reply:
Done :-)